engines/bmc.cpp

```diff
@@ -45,13 +45,13 @@ void Bmc::initialize()
   super::initialize();
 
   logger.log(2, "BMC adding init constraint for step 0");
-  solver_->reset_assertions();
   solver_->assert_formula(unroller_.at_time(ts_.init(), 0));
 }
 
 void Bmc::reset_env()
 {
   initialized_ = false;
+  solver_->reset_assertions();
```

Comment:
I’d prefer putting `reset_assertions()` in `initialize()`, since the solver may also be used externally for other checks. This ensures that the first call to `check_until()` always starts from a clean solver state.

Reply:
Reverted this.